server/api/event/register/[eventUID].delete.ts

```diff
@@ -0,0 +1,61 @@
+// DELETE /api/event/register/:eventUID
+// endpoint for opting out of an event
+export default defineEventHandler(async (event) => {
+  const { user } = event.context
+  const eventUID = getRouterParam(event, 'eventUID') as string
+
+  // user is not authenticated
+  if (!user) {
+    throw createError({
+      statusCode: 404,
```

Comment:
I think the error here should be 401 and not 404, as the situation most likely is that the user did not provide an idToken.